### ISSUE TITLE
[CodeGen]: Enums are serializing as integers from ASP.NET services

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#545]
The JSON returned by ASP.NET services using the Neon extensions include enum values rendered as integers when we expected these to render the string values specified by the `[EnumMember]` attributes.  The strange thing is that these render fine when we serialize the entity locally.

**Comment by jefflill** [comment]
After some investigation, the problem appears only when ASP.NET endpoints are returning a list of round-trip objects.  I suspect that this is happening because our custom converters are being asked whether they can handle a type like `List<TRoundtripData>` and we're answering **no** because we're only handling types that implement `IRoundtripData` and this won't be true for types like `List` or `Dictionary`, etc.

This means we'd have to do a deeper inspection of the types passed and perhaps maintain a dictionary of answers for better performance.

**Comment by jefflill** [comment]
FIXED: I ended up with a simpler and faster solution (even over the older implementation).

I modified the `IMvcBuilder.AddNeon()` extension method to accept an optional lambda function that accepts a `Type` parameter and returns `true` if the type is allowed to be handled by the custom round-trip formatters and I pass this to the formatter constructors.  When this is `null`, the formaters handle all types, otherwise they'll handle only those types for which the lambda function returns `true`.

This is simple, pretty flexible and I also was able to remove a dictionary operations and a couple thread locks to boot.
